corehq/apps/hqwebapp/utils.py

```diff
@@ -165,3 +165,166 @@ def get_bulk_upload_form(context, context_key="bulk_upload"):
+                          'urlname': 'domain_my_account'}],
+            'title': <django.utils.functional.__proxy__ object at 0x106a5c790>,
+            'url': '/a/sravan-test/settings/users/web/'}])]
+    Sample output:
+        [{'data_id': None,
+          'html': None,
+          'is_divider': False,
+          'is_header': True,
+          'title': u'Application Users',
+          'url': None},
+         {'data_id': None,
+          'html': None,
+          'is_divider': False,
+          'is_header': False,
+          'title': u'Mobile Workers',
+          'url': '/a/sravan-test/settings/users/commcare/'},
+         {'data_id': None,
+          'html': None,
+          'is_divider': False,
+          'is_header': False,
+          'title': u'New Mobile Worker',
+          'url': '/a/sravan-test/settings/users/commcare/add_commcare_account/'},
+         {'data_id': None,
+          'html': None,
+          'is_divider': False,
+          'is_header': False,
+          'title': u'Groups',
+          'url': '/a/sravan-test/settings/users/groups/'},
+         {'data_id': None,
+          'html': None,
+          'is_divider': False,
+          'is_header': True,
+          'title': u'Project Users',
+          'url': None},]
+    """
+    dropdown_items = []
+    more_items_in_sidebar = False
+    for side_header, side_list in sidebar_items:
+        dropdown_header = format_submenu_context(side_header, is_header=True)
+        current_dropdown_items = []
+        for side_item in side_list:
+            show_in_dropdown = side_item.get("show_in_dropdown", False)
+            if show_in_dropdown:
+                second_level_dropdowns = get_second_level_dropdowns(
+                    side_item.get('subpages', []), domain=domain)
+                dropdown_item = format_submenu_context(
+                    side_item['title'],
+                    url=side_item['url'],
+                    second_level_dropdowns=second_level_dropdowns,
+                )
+                current_dropdown_items.append(dropdown_item)
+                first_level_dropdowns = get_first_level_dropdowns(
+                    side_item.get('subpages', []), domain=domain
+                )
+                current_dropdown_items = current_dropdown_items + first_level_dropdowns
+            else:
+                more_items_in_sidebar = True
+        if current_dropdown_items:
+            dropdown_items.extend([dropdown_header] + current_dropdown_items)
+
+    if more_items_in_sidebar and current_url_name:
+        return dropdown_items + divider_and_more_menu(current_url_name)
+    else:
+        return dropdown_items
+
+
+def get_second_level_dropdowns(subpages, domain=None):
+    """
+        formats subpages of a sidebar_item as second level dropdown items
+    """
+    second_level_dropdowns = []
+    for subpage in subpages:
+        if (subpage.get('show_in_dropdown', False) and
+           not subpage.get('show_in_first_level', False)):
```

Comment:
Each link in the sidebar (say Mobile Workers) has subpages (Add Mobile Worker for Mobile Workers). These won't be shown in Dropdown menu, but if `show_in_dropdown` is `True` then it's shown as a second-level-dropdown, unless `show_in_first_level` is explicitly set to `True`. So there is no setting called `show_in_second_level`.